client/cgo/verification.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import "C"
+
+import (
+	"bytes"
+	"unsafe"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+// main is required to build a shared library, but does nothing
+func main() {}
+
+//export cgoVerifySignature
+func cgoVerifySignature(pk unsafe.Pointer, pkSize C.int,
+	message unsafe.Pointer, size C.int,
+	sig unsafe.Pointer, sigSize C.int) C.int {
+	if int(pkSize) < sign.PublicKeySize {
```

Comment:
> I guess you're trying to be safe and avoid the chance for an out-of-bounds read.

Yes, since we are dealing with `unsafe.Pointer`. Should I remove these checks?